test/unit/org/apache/cassandra/io/sstable/SSTableReaderTest.java

```diff
@@ -247,6 +253,23 @@ public void testReadRateTracking()
 
         Util.getAll(Util.cmd(store, key).includeRow("0").build());
         assertEquals(2, sstable.getReadMeter().count());
+
+        // With persistence enabled, we should be able to retrieve the state of the meter.
+        sstable.maybePersistSSTableReadMeter();
+
+        UntypedResultSet meter = SystemKeyspace.readSSTableActivity(store.keyspace.getName(), store.name, sstable.descriptor.id);
+        // So this is the only issue atm; seems like it's empty
+        assertFalse(meter.isEmpty());
+
+        Util.getAll(Util.cmd(store, key).includeRow("0").build());
+        assertEquals(3, sstable.getReadMeter().count());
+
+        // After cleaning existing state and disabling persistence, there should be no meter state to read.
+        SystemKeyspace.clearSSTableReadMeter(store.keyspace.getName(), store.name, sstable.descriptor.id);
+        DatabaseDescriptor.setSStableReadRatePersistenceEnabled(false);
```

Comment:
nit: Any risk to any of the other tests if we don't switch back to the original setting?

Reply:
Good catch; I'm going to try/finally wrap it on commit just to make sure it reverts to whatever it entered with. Better safe than sorry.